apollo-router/src/compute_job.rs

```diff
@@ -0,0 +1,78 @@
+use std::future::Future;
```

Comment:
It would be great to see some unit tests for this new module.

Reply:
Added some